neuraxle/pipeline.py

```diff
@@ -50,164 +46,15 @@ def transform(self, data_inputs):
-        """
-        Persist pipeline for current data container
-
-        :param pipeline:
-        :param data_container:
-        :return:
-        """
-        pipeline_cache_folder = os.path.join(self.cache_folder, pipeline.name)
-
-        if not os.path.exists(pipeline_cache_folder):
-            os.makedirs(pipeline_cache_folder)
-
-        next_cached_pipeline_path = self._create_cached_pipeline_path(pipeline_cache_folder, data_container)
-        dump(pipeline, next_cached_pipeline_path)
-
-        return pipeline
-
-    def can_load(self, pipeline: 'Pipeline', data_container: DataContainer) -> bool:
-        """
-        Returns True if the pipeline can be loaded with the passed data container
-
-        :param pipeline:
-        :param data_container:
-        :return: pipeline
-        """
-        pipeline_cache_folder = os.path.join(self.cache_folder, pipeline.name)
-
-        return os.path.exists(
-            self._create_cached_pipeline_path(
-                pipeline_cache_folder,
-                data_container
-            )
-        )
-
-    def load(self, pipeline: 'Pipeline', data_container: DataContainer) -> 'Pipeline':
-        """
-        Load pipeline for current data container.
-
-        :param pipeline:
-        :param data_container:
-        :return: pipeline
-        """
-        pipeline_cache_folder = os.path.join(self.cache_folder, pipeline.name)
-
-        return load(
-            self._create_cached_pipeline_path(
-                pipeline_cache_folder,
-                data_container
-            )
-        )
-
-    def _create_cached_pipeline_path(self, pipeline_cache_folder, data_container: 'DataContainer') -> str:
-        """
-        Create cached pipeline path with data container, and pipeline cache folder.
-
-        :type data_container: DataContainer
-        :param pipeline_cache_folder: str
-
-        :return: path string
-        """
-        all_current_ids_hash = self._hash_data_container(data_container)
-        return os.path.join(pipeline_cache_folder, '{0}.joblib'.format(str(all_current_ids_hash)))
-
-    def _hash_data_container(self, data_container):
-        """
-        Hash data container current ids with md5.
-
-        :param data_container: data container
-        :type data_container: DataContainer
-
-        :return: str hexdigest of all of the current ids hashed together.
-        """
-        all_current_ids_hash = None
-        for current_id, *_ in data_container:
-            m = hashlib.md5()
-            m.update(str.encode(current_id))
-            if all_current_ids_hash is not None:
-                m.update(str.encode(all_current_ids_hash))
-            all_current_ids_hash = m.hexdigest()
-        return all_current_ids_hash
-
 
 class Pipeline(BasePipeline):
     """
     Fits and transform steps
     """
 
-    def __init__(self, steps: NamedTupleList):
+    def __init__(self, steps: NamedTupleList, cache_folder=DEFAULT_CACHE_FOLDER):
```

Comment:
So the cache folder be passed to setup context too and it'd override the provided one a construction, right?

Reply:
yes